### ISSUE TITLE
ragondin-embedder-onnx: the attention mask is built from the id count, so a padding tokenizer pools [PAD] into the mean

### ISSUE THREAD

**Issue by AyRickk** [genesary/Ragondin#237]
### Type / Milestone / Labels
- **Type:** bug
- **Labels:** `type:bug`, `area:components`, `crate:ragondin-embedder-onnx`, `size:S`, `agent-ready`

### Blocked by / Blocks
- **Blocks:** #235 (the SciFact leaderboard calibration cannot pass its 0.5-point tolerance while this stands — it misses by 53 points).

### Context

`ragondin-embedder-onnx` builds the attention mask it pools over from **the number of ids the tokenizer returned**, not from the encoding's own attention mask:

```rust
// src/embedder.rs, embed_batch
for (column, id) in encoding.get_ids().iter().enumerate() {
    ids[row * width + column] = i64::from(*id);
    // Padding keeps the id 0 it was initialized with, and its mask
    // stays 0. The id itself is immaterial twice over: the model's
    // own attention masks it, and the pooling below skips it.
    mask[row * width + column] = 1;
}
```

The comment states the assumption plainly, and the assumption is false for any tokenizer that pads for itself. `tokenizer.json` as sentence-transformers ships it — and as `optimum-cli export onnx` copies it — carries `padding: {"strategy": {"Fixed": 128}}`. `get_ids()` then returns 128 ids for a four-token sentence, of which 124 are `[PAD]`, and every one of them is marked attended. The `[PAD]` hidden states are averaged in as though they were text, and `width` (line 444, also computed from `get_ids().len()`) is the padded width rather than the longest real sequence, so the crate's own "pad the batch to its own longest member" (`ARCHITECTURE.md`) does not hold either.

`ragondin-reranker-onnx` does the same job correctly two lines apart — `src/reranker.rs` writes the ids from `get_ids()` and then the mask from `get_attention_mask()`. The embedder is the one that infers it.

**Why no test catches it.** `tests/fixtures/tokenizer.json` and `tokenizer-bert.json` have no padding configured, so `get_ids().len()` *is* the real length there and the inferred mask is accidentally right. The crate's whole test suite runs inside that blind spot. The defect is only reachable with a real model's tokenizer, which the fixtures deliberately are not.

### The measurement

Found while calibrating the harness against the published SciFact figure (#235), on `sentence-transformers/all-MiniLM-L6-v2` @ `8b3219a92973c328a8e22fadcfa821b5dc75636a`, exported with `optimum-cli export onnx --library-name transformers --task feature-extraction`, over BEIR SciFact (`test` split, 5183 documents, 300 judged queries), dense-only, exact search through `ragondin-store-memory`:

| | nDCG@10 | recall@10 |
|---|---|---|
| MTEB published figure | 0.64508 | — |
| Python, replaying this crate's documented procedure (mean over the real mask, L2, 256 tokens) | **0.6450816** | 0.7833 |
| Python, replaying the mask this crate actually builds | **0.1108616** | 0.2606667 |
| `ragondin bench`, dense-only | **0.1109** | 0.2607 |

The third and fourth rows agree to four decimal places, which is what identifies the mask as the whole of the gap rather than one contributor to it. The second row reproduces the published figure to five, which is what rules out the dataset, the split, the ids, the title rule, the export and the metric.

### Scope — IN
- `components/ragondin-embedder-onnx`: build the mask from `encoding.get_attention_mask()`, and take `width` from the same source, so a tokenizer that pads for itself and one that does not produce the same vector for the same text.
- A fixture tokenizer **with padding configured**, generated by `tests/fixtures/generate.py` beside the existing two, and a test that fails on the current code: the same text embedded under a padding tokenizer and a non-padding one must produce the same vector.

### Scope — OUT
- No change to pooling, normalization, truncation or batching beyond what the mask fix requires.
- No new knob. The mask is not a configuration question.
- No model in the repository and no download at test time — the new fixture tokenizer is generated, like the two beside it.
- No change to `ragondin-reranker-onnx`, which already reads the mask it is given.

### Invariants you must not violate
- **ADR-C14:** the backend stays behind the crate's `onnx` feature.
- The fixtures rule this crate was created under (#20): no model is fetched at build or test time.

### Acceptance criteria
- [ ] A test embeds one text through a padding tokenizer and through a non-padding one and asserts the two vectors are equal; it fails on `main` and passes after the fix.
- [ ] `tests/fixtures/generate.py` regenerates the new tokenizer deterministically, as it does the existing ones.
- [ ] `just check` passes.

### Implementation notes
- `Encoding::get_attention_mask()` returns `&[u32]`, one entry per id; it is already what the reranker uses.
- Worth asserting in the same test that `width` is the longest *real* sequence of the batch, since that is the second thing the id count was standing in for.
- The calibration in #235 is the end-to-end check: with this fixed, dense-only over SciFact should land within 0.5 point of 0.64508. That run belongs to #235 and is not an acceptance criterion here.
